index.d.ts

```diff
@@ -0,0 +1,214 @@
+     */
+    transportSerialDataBits?: 8 | 7 | 6 | 5;
+
+    /**
+     * optional, Must be one of: 1 or 2.
+     */
+    transportSerialStopBits?: 1 | 2;
+
+    /**
+     * optional, Must be one of: 'none', 'even', 'mark', 'odd', 'space'
+     */
+    transportSerialParity?: 'none' | 'even' | 'mark' | 'odd' | 'space';
+
+    /**
+     * optional, default value is 300000 (means after 300000 bytes without a matching message an Error is thrown )
+     */
+    transportSerialMaxBufferSize?: number;
+
+    /**
+     * ms, optional, default value is 120000 (means after 120000ms without a matching message or new data an Error is thrown )
+     */
+    transportSerialMessageTimeout?: number;
+}
+
+interface ObisHttpRequestTransportOptions {
+    /**
+     * required, value SerialResponseTransport, SerialRequestResponseTransport, HttpRequestTransport, LocalFileTransport, StdInTransport or TCPTransport
+     */
+    transport: 'HttpRequestTransport';
+
+    /**
+     * required for HttpRequestTransport, Request URL to query data from
+     */
+    transportHttpRequestUrl: string;
+
+    /**
+     * optional for HttpRequestTransport, Timeout in ms, defaut 2000
+     */
+    transportHttpRequestTimeout?: number;
+}
+
+interface ObisLocalFileTransportOptions {
+    /**
+     * required, value SerialResponseTransport, SerialRequestResponseTransport, HttpRequestTransport, LocalFileTransport, StdInTransport or TCPTransport
+     */
+    transport: 'LocalFileTransport';
+
+    /**
+     * required for LocalFileTransport, File patch to read data from
+     */
+    transportLocalFilePath: string;
+}
+
+interface ObisStdInTransportOptions {
+    /**
+     * required, value SerialResponseTransport, SerialRequestResponseTransport, HttpRequestTransport, LocalFileTransport, StdInTransport or TCPTransport
+     */
+    transport: 'StdInTransport';
+
+    /**
+     * optional, default value is 300000 (means after 300000 bytes without a matching message an Error is thrown )
+     */
+    transportStdInMaxBufferSize?: number;
+
+    /**
+     * ms, optional, default value is 120000 (means after 120000ms without a matching message or new data an Error is thrown )
+     */
+    transportStdInMessageTimeout?: number;
+}
+
+interface ObisTCPTransportOptions {
+    /**
+     * required, value SerialResponseTransport, SerialRequestResponseTransport, HttpRequestTransport, LocalFileTransport, StdInTransport or TCPTransport
+     */
+    transport: 'TCPTransport';
+
+    /**
+     * optional, default value is 300000 (means after 300000 bytes without a matching message an Error is thrown )
+     */
+    transportTcpMaxBufferSize?: number;
+
+    /**
+     * ms, optional, default value is 120000 (means after 120000ms without a matching message or new data an Error is thrown )
+     */
+    transportTcpMessageTimeout?: number;
+}
+
+
+declare abstract class ObisTransport {
+    process(): void;
+
+    stop(): void;
+}
+
+
+export {ObisNames}
```

Comment:
I think you're not using `ObisNames` in this file, so you should just do `export * as ObisNames from "./lib/ObisNames"`.

Reply:
Thanks, that's definitely the cleaner solution!